raster/apply-mosaic-rule-to-rasters/src/main/java/com/esri/samples/apply_mosaic_rule_to_rasters/ApplyMosaicRuleToRastersSample.java

```diff
@@ -0,0 +1,212 @@
+      controlsVBox.setPadding(new Insets(10.0));
+      controlsVBox.setMaxSize(210, 50);
+      controlsVBox.setVisible(false);
+      // add the label and combo box to the control panel
+      controlsVBox.getChildren().addAll(mosaicRuleLabel, comboBox);
+
+      // create a progress indicator
+      ProgressIndicator progressIndicator = new ProgressIndicator(ProgressIndicator.INDETERMINATE_PROGRESS);
+
+      // create a map view
+      mapView = new MapView();
+
+      // add draw status listener to the map view
+      mapView.addDrawStatusChangedListener (drawStatusChangedEvent -> {
+        // show progress indicator while map view is drawing
+        if (drawStatusChangedEvent.getDrawStatus() == DrawStatus.IN_PROGRESS) {
+          progressIndicator.setVisible(true);
+        }
+        else {
+          progressIndicator.setVisible(false);
+        }
+      });
+
+      // create a map with the light gray basemap style
+      ArcGISMap map = new ArcGISMap(BasemapStyle.ARCGIS_LIGHT_GRAY);
+
+      // set the map to the map view
+      mapView.setMap(map);
+
+      // create a raster layer from the image service raster
+      String imageServiceURL = "https://sampleserver7.arcgisonline.com/server/rest/services/amberg_germany/ImageServer";
+      ImageServiceRaster imageServiceRaster = new ImageServiceRaster(imageServiceURL);
+      rasterLayer = new RasterLayer(imageServiceRaster);
+
+      // add raster layer as an operational layer to the map
+      map.getOperationalLayers().add(rasterLayer);
+
+      // listen for the raster layer to finish loading
+      rasterLayer.addDoneLoadingListener(() -> {
+        if (rasterLayer.getLoadStatus() == LoadStatus.LOADED) {
+          // when loaded, set map view's viewpoint to the image service raster's center
+          mapView.setViewpoint(new Viewpoint(imageServiceRaster.getServiceInfo().getFullExtent().getCenter(), 25000.0));
+
+          // enable UI interaction once the raster layer has loaded
+          controlsVBox.setVisible(true);
+
+          // add the mosaic rules to the combo box
+          comboBox.getItems().addAll("Default", "Northwest", "Center", "By attribute", "Lock raster");
+
+          // set the default combo box value
+          comboBox.setValue("Default");
+
+          // create a mosaic rule for the image service raster if one does not already exist
+          if (imageServiceRaster.getMosaicRule() == null) {
+            mosaicRule = new MosaicRule();
+            imageServiceRaster.setMosaicRule(mosaicRule);
+          }
+        } else {
+          // show alert if raster layer fails to load.
+          new Alert(Alert.AlertType.ERROR, "Error loading raster layer.").show();
+        }
+      });
+
+      // set the mosaic rule of the image service raster based on rule chosen from the combo box
+      comboBox.getSelectionModel().selectedItemProperty().addListener(e -> {
+        // create a new mosaic rule
+        mosaicRule = new MosaicRule();
```

Comment:
There's an argument here for checking out whether it's necessary to overwrite our member variable `MosaicRule mosaicRule`, and then explicitly set it `imageServiceRaster.setMosaicRule(mosaicRule);` every time (line 179).
I'm wondering if it'd be possible to set the mosaic rule once when the app starts up, and then just modify the contents of the member variable `MosaicRule mosaicRule`, and the rule would update properly in the view every time? Let's discuss this on a call, it might be easier to explain (and we can test it!)